### ISSUE TITLE
How does that project relate to new upcoming AngularJS 2.0

### ISSUE THREAD

**Issue by Miuler** [jokade/scalajs-angulate#48]
Considering that AngularJS 2.0 is a serial killer
https://twitter.com/mgonto/status/525206705156939776
How does that fit with this project road-map. Will it support / relate to / with AngularJS 2.0 to be more like Dart + Angular 2.0 but with Scala?

**Comment by Miuler** [comment]
Genial, it work now :D

**Comment by jokade** [comment]
I'm not sure about the project road-map and the project's relation to AngularJS 2.0. I've only started to consider Angular 2.0 three months ago, and there was not much information on the concept and implementation details available back then. This has changed with the publication of the [Angular 2.0 API preview](https://angular.io/docs/js/latest/api/).

My first intention was to develop angulate into a migration path from Angular 1.x to Angular 2.0 by back-porting some of the core features of Angular 2.0, and providing a consistent API for angulate that can be used with Angular 1.x and Angular 2.0. That's why I have included a Angular2-style `@Component` annotation with the current `0.2-SNAPSHOT`. However, since there are some profound conceptual changes in Angular2 and the templates are not compatible, I wonder if it's worth the effort.

What's your opinion? Would it be useful to have a library that supports some common functionality that can be compiled to Angular 1.x and Angular 2.x? Or do you think there is no need for migration support, since projects either keep using Angular 1.x or have to be completly re-written for Angular 2.x, anyway?

**Comment by msatala** [comment]
I prefer for angulate to target angular 1.x and when angular 2.x becomes a real thing, create angulate2. There are several reasons for this:
- You are obviously busy and the development of angulate is quite slow. I would prefer bug-free and feature-rich implementation for angular 1.x with shorter development cycle.
- What you are trying to achieve sounds like a good idea, but I think it might never work as you intend. Take @Component annotation for instance. It is clearly better approach and I wanted to use it in my hobby project. However it is incompatible with the rest of the angular 1.x ecosystem so there's not much point. My project (and I think most others as well) is centred around ui-router which only works with controllers.

By the way thank you for your hard work on angulate. In combination with ionic framework it allowed me to create web, mobile, android and ios application with one codebase written in the best language in the world - scala :)

**Comment by jokade** [comment]
> I prefer for angulate to target angular 1.x and when angular 2.x becomes a real thing, create angulate2.

You're right, it's probably best to start a separate project for Angular2, and maybe later a helper library (or sbt plugin) to ease the transition from angulate to angulate2 (if possible).

BTW: I've opened a separate [discussion thread for the project road map](51).

**Comment by Miuler** [comment]
I prefer for angulate to target angular 2.x

**Comment by Miuler** [comment]
Angular 2 is clean, is better than 1.x, and since this project is new, it would be nice to migrate to angular2

**Comment by joaopmafra** [comment]
I prefer for angulate to focus on angular 1.x too. Angular 2 isn't production ready, there are many change possibilities. Focusing on angular 1.x will let people create applications for production now, not for tomorrow.

**Comment by jokade** [comment]
Hi all,
since Angular 2 really breaks with most of the Angular 1.x concepts (and API), I've decided to start a [seaparate project for the Scala.js / Angular 2 binding](https://github.com/jokade/angulate2).  
Maybe there will be a library (or sbt plugin) in the future to ease the migration from angulate to angulate2, but right know these two projects are completly independent.

Cheers, Johannes
